lib/kubeclient/common.rb

```diff
@@ -51,6 +53,9 @@ def initialize_client(
       validate_auth_options(auth_options)
       handle_uri(uri, path)
 
+      @class_owner = class_owner
+      @entities = nil
```

Comment:
That I can do. I would like to keep the same initialization in discover  - so that it purges the list of entities each time it is called.